### PR TITLE
Add benchmarks for `dask.order.order`.

### PR DESCRIPTION
**Pull request by eriknw**
xref #27
xref https://github.com/dask/dask/pull/5646

### CODE REVIEW

File: dask/benchmarks/order.py [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+            prev += (i,)
+
+    def time_order_linear_full(self):
+        order(self.dsk)
+
+
+class OrderManySubgraphs(DaskSuite):
+    """This tests behavior when there are few or many disconnected subgraphs"""
+
+    def setup(self):
```
Comment (TomAugspurger) [inline dask/benchmarks/order.py]: May be able to use a parametrized benchmark here. https://asv.readthedocs.io/en/stable/writing_benchmarks.html#parameterized-benchmarks

I think the only real benefit would be shorter setup times, as only the one `create_disconnected_subgraph` would be called. Possibly nicer reporting.

**Review by TomAugspurger — COMMENTED** [review]
Thanks! Looks good at a glance.

Can you post the timings for these with `asv dev -b order`? Just to make sure times are in the right ballpark.

**Comment by TomAugspurger** [comment]
Looks like we'll need to add scipy as a dep for the linalg one

```
[ 29.27%] ··· order.OrderLinalgSolves.time_order_linalg_solve             failed
[ 29.27%] ···· Traceback (most recent call last):
                 File "/usr/share/miniconda/lib/python3.7/site-packages/asv/benchmark.py", line 1292, in main_run_server
                   main_run(run_args)
                 File "/usr/share/miniconda/lib/python3.7/site-packages/asv/benchmark.py", line 1160, in main_run
                   skip = benchmark.do_setup()
                 File "/usr/share/miniconda/lib/python3.7/site-packages/asv/benchmark.py", line 619, in do_setup
                   result = Benchmark.do_setup(self)
                 File "/usr/share/miniconda/lib/python3.7/site-packages/asv/benchmark.py", line 550, in do_setup
                   setup(*self._current_params)
                 File "/home/vsts/work/1/s/dask/benchmarks/order.py", line 84, in setup
                   [da.linalg.solve(xx[i], xy[i]) for i in range(xx.shape[0])], axis=0
                 File "/home/vsts/work/1/s/dask/benchmarks/order.py", line 84, in <listcomp>
                   [da.linalg.solve(xx[i], xy[i]) for i in range(xx.shape[0])], axis=0
                 File "/home/vsts/work/1/s/dask/.asv/env/38a0f9d29d3f9e627eabb8e5003c7f5a/lib/python3.7/site-packages/dask/array/linalg.py", line 1073, in solve
                   p, l, u = lu(a)
                 File "/home/vsts/work/1/s/dask/.asv/env/38a0f9d29d3f9e627eabb8e5003c7f5a/lib/python3.7/site-packages/dask/array/linalg.py", line 833, in lu
                   import scipy.linalg
               ModuleNotFoundError: No module named 'scipy'
               asv: benchmark failed (exit status 1)
```

**Comment by eriknw** [comment]
Current timings from https://github.com/dask/dask/pull/5646

```
[  2.38%] ··· order.OrderFullLayers.time_order_full_100_20                                      240±0ms
[  4.76%] ··· order.OrderFullLayers.time_order_full_10_1000                                     231±0ms
[  7.14%] ··· order.OrderFullLayers.time_order_full_1_50000                                     306±0ms
[  9.52%] ··· order.OrderFullLayers.time_order_full_2_10000                                     252±0ms
[ 11.90%] ··· order.OrderFullLayers.time_order_full_50000_1                                     286±0ms
[ 14.29%] ··· order.OrderFullLayers.time_order_full_500_2                                       307±0ms
[ 16.67%] ··· order.OrderFullLayers.time_order_full_9999_1                                     83.0±0ms
[ 19.05%] ··· order.OrderLinalgSolves.time_order_linalg_solve                                   274±0ms
[ 21.43%] ··· order.OrderLinearFull.time_order_linear_full                                      827±0ms
[ 23.81%] ··· order.OrderLinearWithDanglers.time_order_linear_danglers_2                        139±0ms
[ 26.19%] ··· order.OrderLinearWithDanglers.time_order_linear_danglers_5                        177±0ms
[ 28.57%] ··· order.OrderManySubgraphs.time_order_many_subgraphs_100_99                         186±0ms
[ 30.95%] ··· order.OrderManySubgraphs.time_order_many_subgraphs_10_999                         253±0ms
[ 33.33%] ··· order.OrderManySubgraphs.time_order_many_subgraphs_1_9999                         165±0ms
[ 35.71%] ··· order.OrderManySubgraphs.time_order_many_subgraphs_30_303                         192±0ms
[ 38.10%] ··· order.OrderManySubgraphs.time_order_many_subgraphs_3_3333                         184±0ms
[ 40.48%] ··· order.OrderManySubgraphs.time_order_many_subgraphs_999_10                         205±0ms
[ 42.86%] ··· order.OrderMapOverlap.time_order_mapoverlap_large                                 2.79±0s
[ 45.24%] ··· order.OrderMapOverlap.time_order_mapoverlap_medium                                100±0ms
[ 47.62%] ··· order.OrderRechunkTranspose.time_order04                                          508±0ms
[ 50.00%] ··· order.OrderSVD.time_order_svd                                                    39.3±0ms
```
`time_order_mapoverlap_large` is the outlier at 2.8 s.

File: dask/benchmarks/order.py (hunk 1 above)
Comment (eriknw) [reply dask/benchmarks/order.py]: Done (and applied elsewhere).

**Pull request merged**